rust/lance-datafusion/src/exec.rs

```diff
@@ -208,14 +209,40 @@ pub fn new_session_context(options: LanceExecutionOptions) -> SessionContext {
     SessionContext::new_with_config_rt(session_config, runtime_env)
 }
 
+lazy_static! {
+    static ref DEFAULT_SESSION_CONTEXT: SessionContext =
+        new_session_context(LanceExecutionOptions::default());
+    static ref DEFAULT_SESSION_CONTEXT_WITH_SPILLING: SessionContext = {
+        new_session_context(LanceExecutionOptions {
+            use_spilling: true,
+            ..Default::default()
+        })
+    };
+}
+
+pub fn get_session_context(options: LanceExecutionOptions) -> SessionContext {
+    let session_ctx: SessionContext;
+    if options.mem_pool_size() == DEFAULT_LANCE_MEM_POOL_SIZE {
```

Comment:
IIUC, if you set a custom memory pool size different from the default, then you will end up creating a new session context each time this is called, won't you?

Reply:
That is correct - that design decision was intentional based on the comment from @westonpace: 

> My thinking is that we will eventually want some kind of process-wide singleton that we use if the caller hasn't requested something special.

I interpreted "something special" as a custom memory pool size. We can make this more generic by introducing a `HashMap`, but we would then need to make the `lazy_static` ref mutable.

Reply:
Got it. Let's create a follow-up issue, but this already seems like an improvement.

Reply:
PR description updated so the implementation is more obvious.